packages/datetime2/src/components/date-input2/dateInput2.tsx

```diff
@@ -88,7 +88,7 @@ export const DateInput2: React.FC<DateInput2Props> = React.memo(function _DateIn
     const handleDateChange = React.useCallback(
         (newDate: Date | null, isUserChange: boolean) => {
             if (newDate == null) {
-                return;
+                return onChange?.(newDate, isUserChange);
```

Comment:
prefer to be even more explicit here:

```suggestion
                onChange?.(null, isUserChange);
                return;
```